classes/migration/upgrade/v3_4_0/I7366_UpdateUserAPIKeySettings.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+use PKP\install\DowngradeNotSupportedException;
+
+class I7366_UpdateUserAPIKeySettings extends \PKP\migration\Migration
```

Comment:
If I'm reading this correctly, it deletes any `apiKeySettings` entries for users who don't have corresponding API keys. The way it's written, it's likely to fail when there are many thousands of users; it's binding too many variables to a single query. I would recommend rewriting this query as a single DELETE.

You'll probably need to rewrite this as a single query, then delete the results in a loop. The query would be something like...
```sql
SELECT u.user_id FROM users u
JOIN user_settings enabled_setting ON (enabled_setting.user_id = u.user_id AND enabled_setting.setting_name = 'apiKeyEnabled')
LEFT JOIN user_settings key_setting ON (key_setting.user_id = u.user_id AND key_setting.setting_name = 'apiKey') 
WHERE key_setting.user_id IS NULL;
```

Reply:
Right as for many thousands for records there will be too many bindings for the `whereIn`. So I went for `chunk` with `non eloquent collection` with a chunk limit `1000` which will reduce delete query . It should able able to handle 1000 data bind in a single query easily .